### PR TITLE
2217 only view submission needed

### LINKED ISSUE #2217: As a 'Data Analyst', I would like to only view sections that need submission
**Description:**
As a 'Data Analyst', I would like to only view sections that I need to submit data files for so that I'm not having to waste time or worry about data files that do not apply to my STT. 

We observed during our v2.0 pilot that certain STTs we held moderated sessions with did not necessarily know what to do with the Section 4 area as they did not have that data file to submit. The knowledge center does have a note about this as a stop gap. 

**Acceptance Criteria:**
- [ ] SSP/TANF radio button continues to drive updates to Reports section below search button
- [ ] Users will not longer see all Sections 1-4 but instead headers based on backend models
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [x] Render the frontend reports page based on backend's model `STT.filenames.keys()` (e.g. ['Active Case Data', 'Closed Case Data', 'Aggregate Data', 'Stratum Data'])
- [x] Ensure frontend unit tests are updated to account for this more dynamic shift for TANF, SSP, and Tribal
- [x] If necessary, ensure the backend filenames.keys are passed to frontend

11/1: Discussed that we would continue to keep the radio button for TANF and SSP to help focus the user on each type

**Notes:**
Scenario: TANF Data Files
Given the STT Data Analyst has been **approved** to submit certain TANF Data Files (Section 1, 2, 3 and/or 4)
When logged into TDP
Then display all applicable sections (Active, Closed, Aggregate and/or Stratum) that the STT Data Analyst has been approved to submit

image
https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/TDP-Mockups-For-Feedback?node-id=7624%3A50249&t=XjNrtEH5J4odxe7L-1

Scenario: SSP Data Files
Given the STT Data Analyst has been **approved** to submit certain STT Data Files (Section 1, 2, 3 and/or 4)
When logged into TDP
Then display all applicable sections (Active, Closed, Aggregate and/or Stratum) that the STT Data Analyst has been approved to submit

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Made `fileUploadSections` redux state to be accessed across different components
- Updated setStt action to get the number of sections the stt needs to submit
- Added tests to cover a valid and invalid stt
- Updated backend Stt view to be a viewset to allow custom actions
Pull request closes #2217 

## How to Test
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in as ofa system admin
2. Change your stt on the datafiles tab and verify you see three or four sections to submit (compare Alabama, California, and any tribe)

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] SSP/TANF radio button continues to drive updates to Reports section below search button
- [x] Users will not longer see all Sections 1-4 but instead headers based on backend models
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if **necessary**

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by elipe17** [comment]
- Update api endpoint to only return a boolean since we know all STTs will submit 3 or 4 sections. Just need to check which one.
- Test updating `pk` in view to `name` for readability

File: tdrs-frontend/src/actions/reports.js [hunk 1]
```diff
@@ -269,8 +270,20 @@ export const SET_SELECTED_YEAR = 'SET_SELECTED_YEAR'
 export const SET_SELECTED_QUARTER = 'SET_SELECTED_QUARTER'
 export const SET_FILE_TYPE = 'SET_FILE_TYPE'
 
-export const setStt = (stt) => (dispatch) => {
-  dispatch({ type: SET_SELECTED_STT, payload: { stt } })
+export const setStt = (stt) => async (dispatch) => {
+  var response = undefined
+  if (stt !== '') {
+    const URL = `${process.env.REACT_APP_BACKEND_URL}/stts/${stt}/num_sections`
```
Comment (jtimpe) [inline tdrs-frontend/src/actions/reports.js]: i still think there's an opportunity to eliminate this api call somehow. `SET_SELECTED_STT` isn't dispatched until the "Search" button is pressed on the datafile page - so not only does it have to reach out for the submitted datafiles/submission history, a second api call needs to be made just for the number of sections

`SET_STTS` is dispatched whenever the frontend retrieves the stt list
image

if each of the stts in the list could have `num_sections`, i think we could avoid this extra api request

Comment (jtimpe) [reply tdrs-frontend/src/actions/reports.js]: i think that could be as simple as modifying the `STTSerializer` on the backend to have a new generated field

Comment (elipe17) [reply tdrs-frontend/src/actions/reports.js]: I see what you're saying, I think that is a better idea! I will make that change!

**Review by jtimpe — APPROVED** [review]
Good stuff! Thanks for the api change 👍

**Review by raftmsohani — APPROVED** [review]
LGTM!

File: tdrs-backend/tdpservice/stts/models.py [hunk 2]
```diff
@@ -39,6 +39,14 @@ class EntityType(models.TextChoices):
     ssp = models.BooleanField(default=False, null=True)
     sample = models.BooleanField(default=False, null=True)
 
+    @property
+    def num_sections(self):
+        """The number of sections this STT submits."""
+        if self.filenames is None:
+            return 4
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/stts/models.py]: Only nit: maybe add constant for returning value

**Comment by ADPennington** [comment]
lgtm! @victoriaatraft @reitermb -- do we need a release note associated with this change? 


_tribe (does not submit section 4)_
STT_tribe_p3
_state that submits SSP and TANF but doesnt submit section 4_
STT_universe_p2
_state that submits SSP and TANF and submits section 4_
STT_sample_p1
_territory that submits section 4_
STT_sample_p4

**Comment by victoriaatraft** [comment]
@ADPennington Unless stt's have been submitting sections they didn't need to in the past and they'd be confused as to why it's not there anymore, I see this as I see this as "corrective" enhancement to the experience that helps people submit what they need to and doesn't need a release note. 
cc @reitermb

**Review by ADPennington — APPROVED** [review]
🚀

**Pull request merged**